pytensor/tensor/rewriting/math.py

```diff
@@ -423,6 +423,108 @@ def local_sumsqr2dot(fgraph, node):
                     return [new_out]
 
 
+@register_specialize
+@node_rewriter([mul, true_div])
+def local_mulexp2expadd(fgraph, node):
+    """
+    This rewrite detects e^x * e^y and converts it to e^(x+y).
+    Similarly, e^x / e^y becomes e^(x-y).
+    """
+    if isinstance(node.op, Elemwise) and isinstance(
+        node.op.scalar_op, (aes.Mul, aes.TrueDiv)
+    ):
```

Comment:
You don't need this first `isinstance` check, the rewrite will only be called to these `Op`s. Other rewrites in the codebase always check at the top, because they predated those changes

Reply:
Oh, gotcha, this is because mul is actually a shorthand for Elemwise{mul,...}, right? Before recognising this I had thought the caller calls this function for any node that looks like */mul/* so I have to do the isinstance(Elemwise) check, but now I get that because of mul's decorator to wrap it into an Elemwise, the the caller only calls this function for nodes that look like */Elemwise/mul/*. At least I hope I now understand it correctly. :)

I made all the changes you requested, so I hope we're good now. Thanks!

Reply:
Yes, you're correct